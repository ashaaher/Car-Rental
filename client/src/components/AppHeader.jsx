import React from 'react';
import { Link } from 'react-router-dom';

import {
  AppBar,
  Tabs,
  Tab,
  Toolbar,
  Typography,
  withStyles,
} from '@material-ui/core';
import { withOktaAuth } from '@okta/okta-react';
import LoginButton from "./LoginButton";
import { withRouter } from 'react-router-dom';

const styles = {
  flex: {
    flex: 1,
  },
  tabLink: {
    display: "flex",
    alignItems: "center",
  }
};

class AppHeader extends React.Component {
  state = {
    authenticated: false,
    user: null,
    role: "admin",
    value: 0,
  };

  componentDidUpdate() {
    this.checkAuthentication();
  }

  componentDidMount() {
    this.checkAuthentication();
  }

  async checkAuthentication() {
    const authenticated = this.props.authState ? this.props.authState.isAuthenticated : false;
    const user = this.props.authService ? await this.props.authService.getUser() : null;
    const role = user && user.groups && user.groups.includes("CarRentalApp Admin") ? "admin": "user";
    if (authenticated !== this.state.authenticated) {
      this.setState({
        authenticated,
        role
      })
    }
  }
  
  handleChange = (event, newValue) => {
    // event.preventDefault();
    this.setState({value: newValue});
  };
  render() {
    return (
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" color="inherit">
            Car Rental System
          </Typography>
          {
            this.state.authenticated ?
              <>
                {
                  this.state.role === "admin" ?
                    <Tabs>
                      <Tab key="profile" label="Home" component={Link} className={this.props.classes.tabLink} to="/"></Tab>
                      <Tab key="vehicletype" label="Vehicle Types" component={Link} className={this.props.classes.tabLink} to="/vehicletype"></Tab>
                      <Tab key="settings" label="Settings" component={Link} className={this.props.classes.tabLink} to="/settings"></Tab>
                      <Tab key="locations" label="Locations" component={Link} className={this.props.classes.tabLink} to="/locations"></Tab>
                      <Tab key="vehicles" label="Vehicles" component={Link} className={this.props.classes.tabLink} to="/vehicles"></Tab>
                      <Tab key="users" label="Users" component={Link} className={this.props.classes.tabLink} to="/users"></Tab>

                    </Tabs> :
                    <Tabs>
                      <Tab key="home" label="Home" component={Link} className={this.props.classes.tabLink} to="/"></Tab>
                      <Tab key="customervehicles" label="Search Vehicles" component={Link} className={this.props.classes.tabLink} to="/customervehicles"></Tab>
                      <Tab key="bookings" label="My Bookings" component={Link} className={this.props.classes.tabLink} to="/bookings"></Tab>
                      <Tab key="profile" label="Profile" component={Link} className={this.props.classes.tabLink} to="/profile"></Tab>
                    </Tabs>
                }
              </>
              : null
          }


          <div className={this.props.classes.flex} />
          <LoginButton />
        </Toolbar>
      </AppBar>
    );
  }

}

export default withRouter(withOktaAuth(withStyles(styles)(AppHeader)));
