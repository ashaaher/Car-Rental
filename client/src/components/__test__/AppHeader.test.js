import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import AppHeader from '../AppHeader';

it('AppHeader Page matches snapshot', () => {
  const wrapper = shallow(<AppHeader />);
  expect(wrapper).toMatchSnapshot();
})