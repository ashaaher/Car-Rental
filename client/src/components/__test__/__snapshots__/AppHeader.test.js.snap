// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader Page matches snapshot 1`] = `ShallowWrapper {}`;
